rfcs/2021-11-whcm-support.md

```diff
@@ -0,0 +1,185 @@
+
+For example, because we've implemented custom checkbox and radio buttons, these elements don't receive all the styles a native checkbox or radio button would normally receive. So when a checkbox or radio button is disabled, that style isn't apparent at all in WHCM. To fix something like this, you'd implement the following:
+
+```css
+@media (-ms-high-contrast: active), (forced-colors: active) {
+  .ds-c-choice:disabled + label {
+    color: GrayText; /* `GrayText` is the CSS System Colors name for disabled text */
+  }
+  .ds-c-choice:disabled + label:before {
+    border-color: GrayText;
+  }
+}
+```
+
+#### Ex. 3: Making sure an element's style persists, regardless of contrast mode
+
+There are some scenarios where you may want to force an element to appear the same in WHCM as you would in any non-WHCM environment. For our purposes, we may want to use this approach for sections of our documentation site, like our [color swatches](https://design.cms.gov/styles/color/), which have visual meaning but don't appear in WHCM due to the limited WHCM color palette.
+
+To preserve the original appearance of an element in WHCM, you could write the following:
+
+```css
+@media (-ms-high-contrast: active), (forced-colors: active) {
+  .c-swatch__preview {
+    -ms-high-contrast-adjust: none; /* For IE */
+    forced-color-adjust: none; /* For everything else * /
+  }
+}
+```
+
+### Recommendation
+
+**WHCM updates should support IE in addition to modern browsers.**
+
+While our design system no longer technically supports IE ([as it falls below our 2% support threshold](https://analytics.usa.gov/health-human-services/)), a good percentage of those who rely on WHCM continue to use it. And because the effort to support IE is low, I agree with Microsoft's recommendation - just be certain your system color keywords work across both queries and the cascade doesn't impact how those styles render (meaning, **test the changes**). If more targeted styles are needed between IE and Edge, the Frankenquery may need to be split.
+
+**New and refactored components should be checked in WHCM as part of our "definition of done".**
+
+We have access to a SauceLabs account and they have [good documentation for how to do local testing](https://docs.saucelabs.com/secure-connections/sauce-connect/quickstart/), so we should use it.
+
+**Collaboration between developer and designer may be needed for some issues, but probably not all.**
+
+There are some issues in WHCM that I would consider low severity and I'd think design could trust development to implement a good solution. I'm thinking border-widths, margin/padding on our buttons, and ensuring System Colors are being implemented correctly (like in the disabled state mentioned above). For these kinds of issues, I can envision a developer making an adequate fix and maybe sharing a screenshot with design if needed.
```

Comment:
In light of Scott's recent point about inclusive language and making sure we aren't forcing designers and engineers into siloed roles, what if you reframed this to be more about what amount of design effort goes into these different kinds of problems. The ones that "design could trust development to implement" are ones that are straightforward and don't require redesigning the component. The ones that require "designer/developer collaboration" are the ones that need to have their designs revisted or require a more thoughtful approach to coming to a solution.

This is not a fully baked thought. It's just something that was in my head, so take it with a grain of salt.

CC @ericwbailey

Reply:
Agreed. I was thinking of @line47's comments here as well, but it was hard for me to articulate generic roles. Ideally, collaboration would happen between those who drive the visual/functional language of our system and those who've implemented it. This could be a collab between designer/developer, product/designer, product/developer, etc. I just don't know how to phrase this at all.

Reply:
I like design/dev collaboration and I like where your head is at with this, @pwolfert!